From 8a715de972062cd905c43b7550b2f92244433c28 Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Sun, 27 May 2018 22:32:11 +0200
Subject: [PATCH 2/2] Jelly: Adapt ProgressBar location based on reach mode

 * Configure ProgressBar and ToolbarSearchBar relative
    position layouts to align the progressbar at the top
    when reach mode is enabled

Change-Id: I77a4544c47f03c18eca99a055c1beb3dbf393073
---
 .../org/lineageos/jelly/MainActivity.java     | 24 ++++++++++++++++---
 1 file changed, 21 insertions(+), 3 deletions(-)

diff --git a/app/src/main/java/org/lineageos/jelly/MainActivity.java b/app/src/main/java/org/lineageos/jelly/MainActivity.java
index be85bfc..079f52b 100644
--- a/app/src/main/java/org/lineageos/jelly/MainActivity.java
+++ b/app/src/main/java/org/lineageos/jelly/MainActivity.java
@@ -75,6 +75,7 @@ import android.widget.ImageButton;
 import android.widget.ImageView;
 import android.widget.LinearLayout;
 import android.widget.ProgressBar;
+import android.widget.RelativeLayout;
 import android.widget.TextView;
 
 import org.lineageos.jelly.favorite.FavoriteActivity;
@@ -132,6 +133,7 @@ public class MainActivity extends WebViewExtActivity implements
     private WebViewExt mWebView;
     private ProgressBar mLoadingProgress;
     private SearchBarController mSearchController;
+    private RelativeLayout mToolbarSearchBar;
     private boolean mHasThemeColorSupport;
     private Drawable mLastActionBarDrawable;
     private int mThemeColor;
@@ -160,6 +162,7 @@ public class MainActivity extends WebViewExtActivity implements
         mAppBar = findViewById(R.id.app_bar_layout);
         mWebViewContainer = findViewById(R.id.web_view_container);
         mLoadingProgress = findViewById(R.id.load_progress);
+        mToolbarSearchBar = findViewById(R.id.toolbar_search_bar);
         AutoCompleteTextView autoCompleteTextView = findViewById(R.id.url_bar);
         autoCompleteTextView.setAdapter(new SuggestionsAdapter(this));
         autoCompleteTextView.setOnEditorActionListener((v, actionId, event) -> {
@@ -430,7 +433,7 @@ public class MainActivity extends WebViewExtActivity implements
     }
 
     private void showSearch() {
-        findViewById(R.id.toolbar_search_bar).setVisibility(View.GONE);
+        mToolbarSearchBar.setVisibility(View.GONE);
         findViewById(R.id.toolbar_search_page).setVisibility(View.VISIBLE);
         mSearchController.onShow();
         mSearchActive = true;
@@ -439,7 +442,7 @@ public class MainActivity extends WebViewExtActivity implements
     @Override
     public void onCancelSearch() {
         findViewById(R.id.toolbar_search_page).setVisibility(View.GONE);
-        findViewById(R.id.toolbar_search_bar).setVisibility(View.VISIBLE);
+        mToolbarSearchBar.setVisibility(View.VISIBLE);
         mSearchActive = false;
     }
 
@@ -768,7 +771,10 @@ public class MainActivity extends WebViewExtActivity implements
                 (CoordinatorLayout.LayoutParams) mAppBar.getLayoutParams();
         CoordinatorLayout.LayoutParams containerParams =
                 (CoordinatorLayout.LayoutParams) mWebViewContainer.getLayoutParams();
-
+        RelativeLayout.LayoutParams progressParams =
+                (RelativeLayout.LayoutParams) mLoadingProgress.getLayoutParams();
+        RelativeLayout.LayoutParams searchBarParams =
+                (RelativeLayout.LayoutParams) mToolbarSearchBar.getLayoutParams();
 
         int margin = (int) UiUtils.getDimenAttr(this, R.style.AppTheme,
                 android.R.attr.actionBarSize);
@@ -776,15 +782,27 @@ public class MainActivity extends WebViewExtActivity implements
         if (isReachMode) {
             appBarParams.gravity = Gravity.BOTTOM;
             containerParams.setMargins(0, 0, 0, margin);
+            progressParams.removeRule(RelativeLayout.ALIGN_PARENT_BOTTOM);
+            progressParams.addRule(RelativeLayout.ALIGN_PARENT_TOP, RelativeLayout.TRUE);
+            searchBarParams.removeRule(RelativeLayout.ABOVE);
+            searchBarParams.addRule(RelativeLayout.BELOW, R.id.load_progress);
         } else {
             appBarParams.gravity = Gravity.TOP;
             containerParams.setMargins(0, margin, 0, 0);
+            progressParams.removeRule(RelativeLayout.ALIGN_PARENT_TOP);
+            progressParams.addRule(RelativeLayout.ALIGN_PARENT_BOTTOM, RelativeLayout.TRUE);
+            searchBarParams.removeRule(RelativeLayout.BELOW);
+            searchBarParams.addRule(RelativeLayout.ABOVE, R.id.load_progress);
         }
 
         mAppBar.setLayoutParams(appBarParams);
         mAppBar.invalidate();
         mWebViewContainer.setLayoutParams(containerParams);
         mWebViewContainer.invalidate();
+        mLoadingProgress.setLayoutParams(progressParams);
+        mLoadingProgress.invalidate();
+        mToolbarSearchBar.setLayoutParams(searchBarParams);
+        mToolbarSearchBar.invalidate();
 
         resetSystemUIColor();
 
-- 
2.17.0

